### ISSUE TITLE
The user can change the email to an registered email after login

### ISSUE THREAD

**Issue by mariana-mxp** [projectwife/mtesitoo-android#82]
Baseline version: 1.0.0
Setup:
- Model: Nexus 4
- Android version: 5.1.1

Pre-conditions: 
- The user 1 has account on tesitoo application.

Steps to reproduce:

1. Start application with user 1.
2. The application shows the first screen.
3. Select the "Profile" option.
4. Change the user email to an registered email of user 2.

Actual behavior:
The application changes the email successfully.

Expect behavior:
The application shows an error messagem informing that Email address is already registered.

Occur: 100% 
Number of execution: 1
Users: 
user 1: mxp3
user 2: mxp

**Comment by mariana-mxp** [comment]
The bug continues to occur in the Apr / 6 / 2017 version.

**Comment by BassMunkee** [comment]
Can confirm that this still happens in the latest build.

**Comment by edu11** [comment]
Hey @ptyagicodecamp  I've tried to get to the root of this bug and it seems to be some problem in the backend. Due to the fact the update profile request includes every field, even when there is no change, I thought that the problem was a mixed response, as some fields could've been updated. However, by debugging request parameters, I isolated the email address and still succeed. In both case, I got the same empty response. 

A note apart (perhaps it's the way it's design, I don't know): profile email gets changed but it doesn't change username/access email, so if I'd want to log in into another device/website, I would still need the old email address to log in.

**Comment by ptyagicodecamp** [comment]
@edu11: David takes care of backend part. He would have better say on the email updating API.

**Comment by edu11** [comment]
Hi @david-r-edgar, would you have some time to have a look at this?

**Comment by david-r-edgar** [comment]
We decided that to simplify user registration and avoid users forgetting their login details, we would use the email as the username.

What this means on the backend is that there are actually two database fields, one for `username` and one for `email`. Similarly in the API, the two fields are treated separately.
The functionality was basically implemented by making the client set the same value for both `username` and `email` in the `POST /vendor/register` request; the server doesn't enforce this in any way.

I investigated, and in the regular web admin interface, neither admin nor vendor is actually able to change the vendor's username. `POST /vendor/profile` basically follows this pattern.
As you've seen, the `email` field is still able to be changed; the API doesn't restrict this.

I see a few options:

1. Don't allow the user or admin to change their username. To make it simple, don't allow them to change their email either. The client could implement this without changing the API.
2. Don't allow the user to change their username. But allow admin to change the username on request. This would require a small amount of work on the server.
3. Change the API and website to support full ability to change username. Probably the client would need some updates too (for example to handle if the username is stored for the autologin?)

In my view, _1_ minimises the amount of development and testing work, and is probably completely satisfactory for v1.
Whereas _3_ requires rather more work, but would likely be better longer term.
